contracts/eosio.system/src/rex.cpp

```diff
@@ -616,6 +616,69 @@ namespace eosiosystem {
 
    }
 
+   /**
+    * @brief Adds returns from the REX return pool to the REX pool
+    */
+   void system_contract::update_rex_pool()
+   {
+      const time_point ct = current_time_point();
+      const auto ret_pool_elem = _rexretpool.begin();
+
+      if ( ret_pool_elem == _rexretpool.end() ||
+           ct < ret_pool_elem->last_update_time + eosio::microseconds{rex_return_pool::dist_interval}) {
+         return;
+      }
+
+      const int64_t    current_rate    = ret_pool_elem->current_rate_of_increase;
+      const uint64_t   time_interval   = ct.time_since_epoch().count() - ret_pool_elem->last_update_time.time_since_epoch().count();
```

Comment:
Instead of using current time, you should be using the highest multiple of `dist_interval` that is less than the current time. This should be used both for determining the time interval as well as updating the `last_distribution_time`.